pillar/environment-demo-public.sls

```diff
@@ -3,9 +3,10 @@ elife_xpub:
         endpoint: https://demo--xpub.elifesciences.org
     orcid:
         # sandbox.orcid.org credentials
-        # client_id: 
+        # client_id:
         # client_secret:
     pubsweet:
         base_url: https://demo--xpub.elifesciences.org
     s3:
         bucket: demo-elife-xpub
+    deployment_target: demo
```

Comment:
We have to add a default in `pillar/elife-xpub.sls` because there are environments that are not `demo` nor `prod`:
- `elife-xpub--ci` (used by https://alfred.elifesciences.org/job/test-elife-xpub/)
- all the `elife-xpub-formula` PR builds, which use temporary instances like `elife-xpub--pr-11-test-...`

Reply:
In general environment names in configurations are a smell, so if it's difficult to think of a default here let's cross-check what this is intended to do - it usually maps to one or more capabilities like `sandbox: on|off`

Reply:
From what I see in https://github.com/elifesciences/elife-xpub/pull/667 it is problematic, as I thought; let's discuss there.